core-web/apps/dotcms-ui/src/app/view/pipes/dot-relative-date/dot-relative-date.pipe.spec.ts

```diff
@@ -1,21 +1,160 @@
-import { Injectable } from '@angular/core';
+import { fakeAsync, tick } from '@angular/core/testing';
 
 import { DotFormatDateService } from '@dotcms/app/api/services/dot-format-date-service';
-
-@Injectable()
-class DotFormatDateServiceMock {
-    getRelative() {
-        return '6 hours ago';
-    }
-}
+import { DotcmsConfigService } from '@dotcms/dotcms-js';
+import { DotcmsConfigServiceMock } from '@dotcms/utils-testing';
 
 import { DotRelativeDatePipe } from './dot-relative-date.pipe';
 
+const MILLISECONDS_IN_A_DAY = 86400000;
+
+const EIGHT_DAYS = MILLISECONDS_IN_A_DAY * 8;
+
+const TWO_DAYS = MILLISECONDS_IN_A_DAY * 2;
+
+const ONE_DAY = MILLISECONDS_IN_A_DAY * 1;
+
+const DATE = '2020/12/3';
+const DATE_AND_TIME = '2020/12/3, 10:08 PM';
+
 describe('DotRelativeDatePipe', () => {
-    it('should set relative date', () => {
-        const formatDateService = new DotFormatDateServiceMock();
+    let formatDateService: DotFormatDateService;
+    let pipe: DotRelativeDatePipe;
+
+    beforeEach(() => {
+        formatDateService = new DotFormatDateService(
+            new DotcmsConfigServiceMock() as unknown as DotcmsConfigService
+        );
+
+        pipe = new DotRelativeDatePipe(formatDateService as unknown as DotFormatDateService);
+    });
+
+    describe('relative', () => {
+        it('should set relative date', fakeAsync(() => {
+            const date = new Date();
+
+            tick(1000);
+
+            expect(pipe.transform(date.getTime())).toEqual('1 second ago');
+        }));
+
+        it('should return relative date even if it is hardcoded date', fakeAsync(() => {
+            const date = DATE;
+            jasmine.clock().mockDate(new Date(DATE));
+
+            jasmine.clock().tick(TWO_DAYS);
+
+            expect(pipe.transform(date)).toEqual('2 days ago');
+        }));
+
+        it('should return relative date even if it is hardcoded date with time', fakeAsync(() => {
+            const date = DATE_AND_TIME;
+
+            jasmine.clock().mockDate(new Date(DATE_AND_TIME));
+
+            jasmine.clock().tick(ONE_DAY);
+
+            expect(pipe.transform(date)).toEqual('1 day ago');
+        }));
+
+        it('should call dotFormatService.getUTC 3 times when its not a timestamp', () => {
+            const date = new Date();
+
+            spyOn(formatDateService, 'getUTC').and.callThrough();
+
+            pipe.transform(date.getTime()); // This is milliseconds not a timestamp
+
+            expect(formatDateService.getUTC).toHaveBeenCalledTimes(3);
+        });
```

Comment:
I feel we don't need to tests this, just need to make sure that the transform is returning what we expect.

Reply:
Alright! I'll delete it